server/app/services/geo/esri/EsriClient.java

```diff
@@ -60,10 +69,26 @@ public EsriClient(Config configuration, WSClient ws) {
         configuration.hasPath("esri_find_address_candidates_url")
             ? Optional.of(configuration.getString("esri_find_address_candidates_url"))
             : Optional.empty();
-    this.ESRI_FIND_ADDRESS_CANDIDATES_TRIES =
-        configuration.hasPath("esri_find_address_candidates_tries")
-            ? configuration.getInt("esri_find_address_candidates_tries")
+    this.ESRI_EXTERNAL_CALL_TRIES =
+        configuration.hasPath("esri_external_call_tries")
+            ? configuration.getInt("esri_external_call_tries")
             : 3;
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS =
+        configuration.hasPath("esri_address_service_area_validation_labels")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_labels"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_VALUES =
+        configuration.hasPath("esri_address_service_area_validation_values")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_values"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS =
+        configuration.hasPath("esri_address_service_area_validation_urls")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_urls"))
+            : Optional.empty();
+    this.ESRI_ADDRESS_SERVICE_AREA_VALIDATION_PATHS =
+        configuration.hasPath("esri_address_service_area_validation_paths")
+            ? Optional.of(configuration.getList("esri_address_service_area_validation_paths"))
+            : Optional.empty();
```

Comment:
**suggestion (blocking):** Please move loading this config data into its own class that can be injected elsewhere in the code. We'll need to access these values in other places such as the predicate editor the admin.

Reply:
@bion , I created a EsriServiceAreaValidationConfig class with some methods to make it more accessible elsewhere and updated EsriClient to use it.

Reply:
Thanks!